### ISSUE TITLE
Hotcakes Sitemap provider doesn't support localization

### ISSUE THREAD

**Issue by Navilois** [HotcakesCommerce/hotcakes-commerce-core#269]
## Describe the bug
Sitemaps generated by the productsSitemapProvider provide incorrect paths when using localized page names. The generated URLs lead to the EN-US product overview instead of the product. This is very problematic for SEO reasons.

Example:
Localized URL: https://www.sierzega.com/it-it/prodotti/vedere-i-prodotti/gr33l-cl
Sitemap URL: https://www.sierzega.com/it-it/products/product-viewer/gr33l-cl

## Software Versions
- DNN:  09.04.04
- Hotcakes:  03.02.03

## To Reproduce
1. Go to https://www.sierzega.com/it-it/sitemap.aspx
2. View page source
3. See error

## Expected behavior
productsSitemapProvider should use localized page names

## Actual behavior
productsSitemapProvider uses en-us (or default language) page names

## Additional context
The DNN coreSitemapProvider works fine and uses the translated pages, thus everthing but the product URLs are correct.

**Comment by Navilois** [comment]
In case you don't have time for this, please point me into the right direction and I'll se if I can fix it.

**Comment by hismightiness** [comment]
I'd have to take some time to debug this to give you more specific direction, but this class is where you'd want to start.

https://github.com/HotcakesCommerce/hotcakes-commerce-core/blob/master/Libraries/Hotcakes.Commerce.Dnn/Providers/ProductsSitemapProvider.cs

**Comment by Navilois** [comment]
I've managed to create a working localized sitemap, unfortunately it's just using my current portal locale when hitting the "Clear Cache" Button at SEO -> Sitemap Settings for all localized sitemaps.

I used `HccRequestContextUtils.UpdateUserContentCulture(HccRequestContext.Current);`
before https://github.com/HotcakesCommerce/hotcakes-commerce-core/blob/47e2271cb231754ae9bbf7045156189552aeea55/Libraries/Hotcakes.Commerce.Dnn/Mvc/DnnHccUrlResolver.cs#L202

I don't know how to get the correct locale in there. DNN does it using PortalSettings.CultureCode.
See lines 54 - 62 here: https://github.com/dnnsoftware/Dnn.Platform/blob/b96c4913c6bcca688e987631a17a81ad2e61905d/DNN%20Platform/Library/Services/Sitemap/CoreSitemapProvider.cs#L54

I'll be back on it on monday. If you have any clues for me I'd be grateful of course ;)

**Comment by hismightiness** [comment]
It sounds like you're on the right path to getting this done.  I'd suggest comparing what you're doing to the underlying base classes in the Hotcakes admin and front-end views to get and apply the locale.

**Comment by Navilois** [comment]
I didn't find a good way to implement the fix within https://github.com/HotcakesCommerce/hotcakes-commerce-core/blob/master/Libraries/Hotcakes.Commerce.Dnn/Providers/ProductsSitemapProvider.cs 
My preferred solution would have been to pass the locale directly at GetPageUrl to RouteHccUrl via the routeValues object.

Instead I solved it by using your HccRequestContextUtils.GetContextWithCulture() method.

I replaced https://github.com/HotcakesCommerce/hotcakes-commerce-core/blob/47e2271cb231754ae9bbf7045156189552aeea55/Libraries/Hotcakes.Commerce.Dnn/Mvc/DnnHccUrlResolver.cs#L202
by
`HccRequestContext localizedContext = HccRequestContextUtils.GetContextWithCulture(HccRequestContext.Current, CurrentPortalSettings.CultureCode);
                urlStoreSettings = localizedContext.CurrentStore.Settings.Urls;`

I'm not 100% certain of the impact of this change nor do I have the automated testing capabilities.  GetTabId() has a very general purpose and I have no idea, if I'm messing up something else. It seems to be working fine so far, but I didn't do a lot of testing yet.

Can you take a closer look?

**Comment by Navilois** [comment]
Wasn't happy with the first solution, this one is much better ;) It's simply localizing the HccRequestContext when the sitemap is generated. I did some testing - works perfectly fine and I don't see a reason this could have an unforseen negative impact.

It's live on my production environment already - e.g. https://www.sierzega.com/es-es/sitemap.xml

**Comment by Navilois** [comment]
Wasn't happy with the first solution, this one is much better ;) It's simply localizing the HccRequestContext when the sitemap is generated. I did some testing - works perfectly fine and I don't see a reason this could have an unforseen negative impact.

It's live on my production environment already - e.g. https://www.sierzega.com/es-es/sitemap.xml


To be discussed:
At the moment, the provider generates URLs for all products. I prefer products that are AvailableForSale = true only, but I removed it (see commit [e9733b5](https://github.com/Navilois/hotcakes-commerce-core/commit/e9733b53106289762fee3e7fd12a1e7369ed020e) again for this pull request since it might be a breaking change for some.
